nova_vm/src/ecmascript/execution/realm.rs

```diff
@@ -96,14 +95,13 @@ impl<'ctx, 'host> Realm<'ctx, 'host> {
         let this_value = this_value.unwrap_or(global_object);
 
         // 4. Set realmRec.[[GlobalObject]] to globalObj.
-        self.global_object = global_object;
+        agent.heap.get_realm_mut(realm_id).global_object = global_object;
```

Comment:
**thought:** I don't know if it would be useful but we might benefit from eg. having `get_realm_mut` and similar kind of APIs at the `agent.heap.realms` level, and having Environments related APIs at the `agent.heap.environments` level etc.

The reason being that then it would be possible to "split up" the borrow of the `agent` like this:
```rs
let &mut Heap {
  realms,
  environments,
  objects,
  ...
} = agent.heap;
```

Now all of those can be individually borrowed at the same time. As long as the internal APIs do not require access to the other parts of the agent / heap then they can be used this way.

But that's not a thing we need to worry about right now :)

Reply:
Interesting, that could help out. I meant for `get_realm` to be synonymous with [_the current Realm Record_](https://tc39.es/ecma262/#current-realm) in the specification. However, we could get away with accessing the heap after that. If only we could force functions like these to be inlined like macros so that borrows could be checked more precisely.